pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -1506,4 +1506,55 @@ self: super: {
   # through $PATH but they aren't in $PATH
   dhall-lsp-server = dontCheck super.dhall-lsp-server;
 
+  # hasn't bumped upper bounds
+  floskell = markUnbroken (dontCheck (doJailbreak super.floskell));
+
+  # hasn't bumped upper bounds
+  monad-dijkstra = markUnbroken (dontCheck (doJailbreak super.monad-dijkstra));
+
+  # packages with the right version, required for haskell-language-server
```

Comment:
For getting packages with the right version, you should pin those in `configuration-hackage2nix.yaml` in the extra-packages list.

Then they would automatically be added to hackage-packages.nix when hackage2nix is run.

You can run hackage2nix yourself by cloning https://github.com/nixos/cabal2nix and using the regenerate-nixpkgs.sh script. (Please tell me if the documentation in the script is unclear.)